source/DoorModel.h

```diff
@@ -0,0 +1,164 @@
+	unsigned char playersOn;
+
+   public:
+#pragma mark Constructors
+	/*
+	 * Creates a new door at angle 0.
+	 *
+	 * NEVER USE A CONSTRUCTOR WITH NEW. If you want to allocate a model on
+	 * the heap, use one of the static constructors instead.
+	 */
+	DoorModel(void) : angle(0), playersOn(0) {}
+
+	/**
+	 * Destroys this door, releasing all resources.
+	 */
+	~DoorModel(void) { dispose(); }
+
+	/**
+	 * Disposes all resources and assets of this door
+	 *
+	 * Any assets owned by this object will be immediately released.  Once
+	 * disposed, a door may not be used until it is initialized again.
+	 */
+	void dispose();
+
+	/**
+	 * Initializes a new door at an unassigned angle (-1).
+	 *
+	 * An initializer does the real work that the constructor does not.  It
+	 * initializes all assets and makes the object read for use.  By separating
+	 * them, we allow ourselfs non-pointer references to complex objects.
+	 *
+	 * @return true if the obstacle is initialized properly, false otherwise.
+	 */
+	virtual bool init() { return init(-1.0f); }
+
+	/**
+	 * Initializes a new door with the given angle
+	 *
+	 * An initializer does the real work that the constructor does not.  It
+	 * initializes all assets and makes the object read for use.  By separating
+	 * them, we allow ourselfs non-pointer references to complex objects.
+	 *
+	 * @param a   The angle at which the door exists
+	 *
+	 * @return true if the obstacle is initialized properly, false otherwise.
+	 */
+	virtual bool init(const float a) {
+		this->angle = a;
+		return true;
+	};
+
+	static std::shared_ptr<DoorModel> alloc() {
+		std::shared_ptr<DoorModel> result = std::make_shared<DoorModel>();
+		return (result->init() ? result : nullptr);
+	}
+
+#pragma mark -
+#pragma mark Accessors
+	/**
+	 * Returns the current angle of the door in radians.
+	 *
+	 * @return the current angle of the door in radians.
+	 */
+	float getAngle() { return (float)M_PI * angle / 180.0f; }
+
+	/**
+	 * Returns the current height of the door.
+	 *
+	 * @return the current height of the door.
+	 */
+	int getHeight() { return height; }
+
+	/**
+	 * Returns the number of players in range of the door.
+	 *
+	 * @return the number of players in range of the door.
+	 */
+	int getPlayersOn() {
+		std::bitset<8> ids(playersOn);
+		return ids.count();
+	}
+
+	/**
+	 * Sets the current angle of the door in radians.
+	 *
+	 * @param value The door angle in radians
+	 */
+	void setAngle(float value) { angle = 180.0f * value / (float)M_PI; }
+
+	/**
+	 * Adds the given player's flag from the door.
+	 *
+	 */
+	void addPlayer(int id) { playersOn = playersOn | (unsigned char)pow(2, id); }
+
+	/**
+	 * Removes the given player's flag from the door. Requires that this player is on the door
+	 *
+	 */
+	void removePlayer(int id) {
+		if (!resolved()) {
+			playersOn = playersOn ^ (unsigned char)pow(2, id);
+		}
+	}
+
+	/**
+	 * Raises the door.
+	 *
+	 */
+	void raiseDoor() {
+		if (height < MAX_HEIGHT) height += SPEED;
```

Comment:
it's only going to be a couple pixels difference depending on SPEED. In the current scenario, since SPEED divides MAX_HEIGHT, height is never greater than MAX_HEIGHT